buildkite/provider.go

```diff
@@ -94,8 +94,9 @@ func (tf *terraformProvider) Metadata(ctx context.Context, req provider.Metadata
 func (*terraformProvider) Resources(context.Context) []func() resource.Resource {
 	return []func() resource.Resource{
 		NewAgentTokenResource,
-		newClusterResource,
+		NewClusterAgentTokenResource,
```

Comment:
Nit: Functions and variables in this don't need to be capitalised, for the most part. They're not exported from the package and so can remain internal.

Reply:
I would like us to do more idiomatic go like this 👍
There is so much already that we need to clean up as well. Lets just lump that onto future us to tidy up at some point